db/write_thread.cc

```diff
@@ -268,6 +268,12 @@ size_t WriteThread::EnterAsBatchGroupLeader(
       break;
     }
 
+    if (!w->no_sleep && leader->no_sleep) {
```

Comment:
I think we need to break for the other way around too.

In the disableWAL case, a query disabling WAL can go with one without, but it's not the case for no waiting.